5.analyze-data/nbconverted/log_reg_class_balanced_areashape_analyze_well_predicted_probabilities.py

```diff
@@ -0,0 +1,165 @@
+# This comparison is accomplished with a KS Test.
+
+# In[ ]:
+
+
+import pathlib
+import sys
+
+import pandas as pd
+from scipy.stats import kstest
+
+# Import significance test utils
+sys.path.append("utils")
+import well_significance_testing as sig_test
+
+
+# ## Find the root of the git repo on the host system
+
+# In[ ]:
+
+
+# Get the current working directory
+cwd = pathlib.Path.cwd()
+
+if (cwd / ".git").is_dir():
+    root_dir = cwd
+
+else:
+    root_dir = None
+    for parent in cwd.parents:
+        if (parent / ".git").is_dir():
+            root_dir = parent
+            break
+
+# Check if a Git root directory was found
+if root_dir is None:
+    raise FileNotFoundError("No Git root directory found.")
+
+
+# ## Input and Output Paths
+
+# In[ ]:
+
+
+# Input paths
+big_drive_path = f"{root_dir}/big_drive"
+sqlite_data_path = f"{big_drive_path}/sc_data"
+
+# Probability data path
+proba_path = pathlib.Path(f"{big_drive_path}/cameron-request__cell-health-plate-classifications/multi_class_models")
+
+# Probability data for each type of model
+final_proba_path = proba_path / "final__CP_areashape_only__balanced"
+shuf_proba_path = proba_path / "shuffled_baseline__CP_areashape_only__balanced"
+
+# Platemap metadata
+platemapdf = pd.read_csv(f"{root_dir}/0.image-download/manifest/idr0080-screenA-annotation.csv")
+
+# Load probability data for final and shuffled models
+final_probadf = pd.concat([pd.read_csv(data_file, index_col=0) for data_file in list(final_proba_path.glob("*.csv.gz"))])
+shuf_probadf = pd.concat([pd.read_csv(data_file, index_col=0) for data_file in list(shuf_proba_path.glob("*.csv.gz"))])
+
+# Output paths
+comparison_results_output_filename = "class_balanced_well_log_reg_areashape_greg_model_comparisons.parquet"
+output_path = pathlib.Path("class_balanced_well_log_reg_comparison_results")
+output_path.mkdir(parents=True, exist_ok=True)
+
+
+# ## Combine shuffled and final model data
+
+# In[ ]:
+
+
+# Define the type of model
+final_probadf["Metadata_Model_Type"] = "final"
+shuf_probadf["Metadata_Model_Type"] = "shuffled"
+
+probadf = pd.concat([final_probadf, shuf_probadf])
+
+
+# ## Merge the platemap and probability data
+
+# In[ ]:
+
+
+platemap_metacols = ["Plate", "Well"]
+platemap_cols = ["Reagent Identifier", "Characteristics [Cell Line]", "Control Type"]
+probadf = probadf.merge(platemapdf[platemap_cols + platemap_metacols], how="inner", left_on=["Metadata_Plate", "Metadata_Well"], right_on=platemap_metacols)
+
+# Drop Redundant columns from merge
+probadf.drop(columns=platemap_metacols, inplace=True)
+
+
+# ## Define phenotype and columns to group by
+
+# In[ ]:
+
+
+phenotype_cols = probadf.loc[:, "ADCCM":"SmallIrregular"].columns.tolist()
+
+filt_cols = ['Metadata_Plate', 'Reagent Identifier', 'Metadata_Model_Type', 'Characteristics [Cell Line]', 'Metadata_Well']
+
+
+# ## KS test wrapper function
+
+# In[ ]:
+
+
+def perform_ks_test(_dmso_probs, _treatment_probs):
```

Comment:
Consider moving the functions to the top of the notebook.

Reply:
alternatively this could be added to the helper file as as helper funciton

Reply:
Thanks for the suggestion, I added this to the top just before where the paths are defined